tests/unit/package_managers/bundler/test_parser.py

```diff
@@ -180,3 +180,32 @@ def test_parse_gemlock_empty(
 
     assert f"Package {rooted_tmp_path.path.name} is bundled with version 2.5.10" in caplog.messages
     assert result == []
+
+
+@mock.patch("cachi2.core.package_managers.bundler.parser.Repo.clone_from")
+def test_download_git_repository(
+    mock_git_clone: mock.Mock,
+    rooted_tmp_path: RootedPath,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    dep = GitDependency(
+        name="example",
+        version="0.1.0",
+        url="https://github.com/user/repo.git",
+        ref="f" * 40,
+    )
+
+    dep_path = rooted_tmp_path.join_within_root(f"{dep.repo_name}-{dep.ref[:12]}").path
+
+    dep.download_to(output_dir=rooted_tmp_path)
+    assert f"Downloading git repository {dep.url}" in caplog.messages
+
+    dep.download_to(output_dir=rooted_tmp_path)
+    assert f"Skipping existing git repository {dep.url}" in caplog.messages
```

Comment:
I would recommend splitting this test into two: you are testing two scenarios here, one that something could be downloaded, another that duplicates are skipped. When such compound tests fail it is harder to pinpoint underlying issue because multiple things are tested simultaneously. Compound tests make sense in integration or e2e where tests are expensive, for UTs I think clarity is more important.